students/JoeNunnelley/session03/slicing_lab.py

```diff
@@ -0,0 +1,101 @@
+#! /usr/bin/env python3
+"""
+Write some functions that take a sequence as an argument, and return
+a copy of that sequence:
+
+    - with the first and last items exchanged.
+    - with every other item removed.
+    - with the first 4 and the last 4 items removed, and then every
+      other item in the remaining sequence.
+    - with the elements reversed (just with slicing).
+    - with the last third, then first third, then the middle third
+      in the new order.
+Author : Joe Nunnelley
+"""
+def exchange_first_last(seq):
+    """1. exchange the first and last items in any sequence"""
+    return seq[-1::] + seq[1:-1] + seq[0:1]
+
+
+def every_other_item_removed(seq):
+    """2. remove every other item from a sequence"""
+    return seq[0::2]
+
+
+def remove_four_bookends_and_every_other(seq):
+    """3. remove 4 on each end and return every other item"""
+    internal = seq[4:-4]
+    if internal:
+        return every_other_item_removed(internal)
```

Comment:
kinda slick! -- I'm sure you know that you could have done: seq[4:-4:2], but I like this.